scripts/docker/astrobee_quick.Dockerfile

```diff
@@ -0,0 +1,42 @@
+# This will set up an Astrobee melodic docker container using the non-NASA
+# install instructions. This image builds on top of the base melodic image
+# building the code. You must set the docker context to be the repository root
+# directory
+#
+# This dockerfile is an alternative to `astrobee.Dockerfile`. It uses previous
+# docker image build artifacts to speed up the catkin build step. If, in the
+# future, you were to use a 'quick' astrobee's image as a build cache, be sure
+# to add `catkin clean --orphans` so you don't bloat the cache image down the
+# line.
+
+ARG UBUNTU_VERSION=16.04
+ARG REMOTE=astrobee
+ARG REMOTE_CACHED=${REMOTE}
+
+FROM ${REMOTE_CACHED}/astrobee:latest-ubuntu${UBUNTU_VERSION} AS build_cache
+FROM ${REMOTE}/astrobee:base-latest-ubuntu${UBUNTU_VERSION}
+
+ARG ROS_VERSION=kinetic
+
+COPY --from=build_cache /src/astrobee/build /src/astrobee/build/
+COPY --from=build_cache /src/astrobee/devel /src/astrobee/devel/
+COPY --from=build_cache /src/astrobee/.catkin_tools /src/astrobee/.catkin_tools/
+
+COPY . /src/astrobee/src/
+
+# reset the timestamps of the files to when they were committed, then cakin will only build new stuff
+# https://stackoverflow.com/questions/2458042/restore-a-files-modification-time-in-git
+RUN cd /src/astrobee/src/ \
+    && \
+    for f in $(git ls-tree -r -t --full-name --name-only "HEAD") ; do \
```

Comment:
This is pretty cool, didn't know about this. I will include it in my future ROS+Docker builds.
One question: if you added .git to the .dockerignore, how does this bash command still work?

Reply:
it's the opposite: _.git must not be ignored_